File: LeetCode75/ArrayAndString/1431_kids_with_the_greatest_number_of_candies/kidsWithCandies.py
from typing import List


class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        candies_max = max(candies)
        res = []

        for i in candies:
            if i + extraCandies >= candies_max:
                res.append(True)
            else:
                res.append(False)

        return res

File: LeetCode75/1431_kids_with_the_greatest_number_of_candies/kidsWithCandies.go
func kidsWithCandies(candies []int, extraCandies int) []bool {
	res := []bool{}
	maxNum := max(candies)

	for _, i := range candies {
		if i+extraCandies >= maxNum {
			res = append(res, true)
		} else {
			res = append(res, false)
		}
	}

	return res
}

func max(list []int) int {
	maxNum := 0

	for _, i := range list {
		if i > maxNum {
			maxNum = i
		}
	}

	return maxNum
}
